spring-cloud-function-context/src/main/java/org/springframework/cloud/function/context/catalog/SimpleFunctionRegistry.java

```diff
@@ -872,9 +876,14 @@ else if (!(input instanceof Publisher) && input instanceof Iterable && !Function
 			return input;
 		}
 
-		/*
-		 *
-		 */
+		private String contentTypeHeaderValue(Message<?> msg) {
+			var contentType = msg.getHeaders().get("contentType");
+			if (contentType == null) {
+				contentType = msg.getHeaders().get("content-type");
+			}
+			return Objects.toString(contentType);
```

Comment:
Note that `Objects.toString` is null safe and will return `"null"` when its input arg is null.